test/utility.h

```diff
@@ -98,6 +98,13 @@ class TestContext : public ContextBase {
 
   bool isLogged(std::string_view message) { return log_.find(message) != std::string::npos; }
 
+  uint64_t getCurrentTimeNanoseconds() override {
+    return std::chrono::system_clock::now().time_since_epoch().count();
+  }
+  uint64_t getMonotonicTimeNanoseconds() override {
+    return std::chrono::steady_clock::now().time_since_epoch().count();
+  }
+
```

Comment:
I know you're only moving the existing code, but as @mpwarres pointed out recently, the nanosecond-resolution is implementation-specific and not guaranteed by the spec spec, so could you wrap both calls with `std::chrono::duration_cast<std::chrono::nanoseconds>()`?